src/lib/operators/join_hash.cpp

```diff
@@ -403,24 +416,30 @@ class JoinHash::JoinHashImpl : public AbstractJoinOperatorImpl {
       hash_tables = build<BuildColumnType, HashedType>(radix_build_column, JoinHashBuildMode::AllPositions, _radix_bits,
                                                        probe_side_bloom_filter);
     }
+    _performance.set_step_runtime(OperatorSteps::Building, timer_hash_map_building.lap());
 
-    // Short cut for AntiNullAsTrue
-    //   If there is any NULL value on the build side, do not bother probing as no tuples can be emitted
-    //   anyway (as long as JoinHash/AntiNullAsTrue doesn't support secondary predicates). Doing this early out
-    //   right here is hacky, but during probing we assume NULL values on the build side do not matter, so we'd have no
-    //   chance detecting a NULL value on the build side there.
+    /**
+     * Short cut for AntiNullAsTrue:
+     *   If there is any NULL value on the build side, do not bother probing as no tuples can be emitted anyway (as
+     *   long as JoinHash/AntiNullAsTrue doesn't support secondary predicates). Doing this early out right here is
+     *   hacky, but during probing we assume NULL values on the build side do not matter, so we'd have no chance
+     *   detecting a NULL value on the build side there.
+     */
     if (_mode == JoinMode::AntiNullAsTrue) {
       for (const auto& build_side_partition : radix_build_column) {
         for (const auto null_value : build_side_partition.null_values) {
           if (null_value) {
-            return _join_hash._build_output_table({});
+            Timer timer_output_writing;
+            const auto result = _join_hash._build_output_table({});
+            _performance.set_step_runtime(OperatorSteps::OutputWriting, timer_output_writing.lap());
```

Comment:
Aren't you overwriting the value here?

Reply:
If I don't get this code wrong, it's returning directly afterwards. So it's not executed a second time.

Reply:
Yepp, my fault. I'd still add an assertion that prevents overriding the stored step runtime.